Symfony/CS/Fixer/PSR2/ClassDefinitionFixer.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\PSR2;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * Fixer for part of the rules defined in PSR2 ¶4.1 Extends and Implements.
+ *
+ * @author SpacePossum
+ */
+final class ClassDefinitionFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+        for ($index = $tokens->getSize() - 1; $index > 0;--$index) {
+            if (!$tokens[$index]->isClassy()) {
+                continue;
+            }
+
+            $classStart = $tokens->getNextTokenOfKind($index, array('{'));
+            $this->fixClassDefinition($tokens, $index, $classStart);
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $start  Class definition token start index
+     * @param int    $end    Class definition token end index
+     */
+    private function fixClassDefinition(Tokens $tokens, $start, $end)
+    {
+        $implements = $tokens->findGivenKind(T_IMPLEMENTS, $start, $end);
+        // check if there is a `implements`  part in the definition, since there are rules for it in PSR 2.
+        if (count($implements) > 0) {
```

Comment:
not sure how decomposition can be done in a meaningful way, there is not code repetition afaik.